### PR TITLE
RateLimitedRetries: Fix code documentation refresh with details

### PR DESCRIPTION
**Pull request by kirankumarkolli**
RateLimitedRetries: fix code documentation refresh with details

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 1]
```diff
@@ -197,16 +197,50 @@ public ConnectionMode ConnectionMode
...
-        /// Get or set the number of times client should retry on rate throttled requests.
+        /// Gets or sets the maximum number of retries in the case where the request fails
+        /// because the Azure Cosmos DB service has applied rate limiting on the client.
         /// </summary>
+        /// <value>
+        /// The default value is 9. This means in the case where the request is rate limited,
+        /// the same request will be issued for a maximum of 10 times to the server before
+        /// an error is returned to the application. If the value of this property is set to 0,
+        /// there will be no automatic retry on rate limiting requests from the client and the exception
+        /// needs to handled at the application level.
```
Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs:208]: NIT needs to **be** handled

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 2]
```diff
@@ -197,16 +197,50 @@ public ConnectionMode ConnectionMode
...
+        /// For an example on how to set this value, please refer to <see cref="ConnectionPolicy.RetryOptions"/>.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs:209]: Remove this line? ConnectionPolicy is internal.

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 3]
```diff
@@ -197,16 +197,50 @@ public ConnectionMode ConnectionMode
...
+        /// </value>
+        /// <remarks>
+        /// <para>
+        /// When a client is sending requests faster than the allowed rate,
+        /// the service will return HttpStatusCode 429 (Too Many Request) to rate limit the client. The current
+        /// implementation in the SDK will then wait for the amount of time the service tells it to wait and
+        /// retry after the time has elapsed.
+        /// </para>
+        /// <para>
+        /// For more information, see <see href="https://docs.microsoft.com/azure/documentdb/documentdb-performance-tips#429">Handle rate limiting/request rate too large</see>.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs:219]:
```suggestion
        /// For more information, see <see href="https://docs.microsoft.com/azure/cosmos-db/performance-tips#429">Handle rate limiting/request rate too large</see>.
```

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 4]
```diff
@@ -197,16 +197,50 @@ public ConnectionMode ConnectionMode
...
+        /// </remarks>
         /// <seealso cref="CosmosClientBuilder.WithThrottlingRetryOptions(TimeSpan, int)"/>
         public int? MaxRetryAttemptsOnRateLimitedRequests { get; set; }
 
         /// <summary>
-        /// Get or set the max time to client is allowed to retry on rate throttled requests. 
+        /// Gets or sets the maximum retry time in seconds for the Azure Cosmos DB service.
         /// </summary>
+        /// <value>
+        /// The default value is 30 seconds. For an example on how to set this value, please refer to <see cref="ConnectionPolicy.RetryOptions"/>.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs:229]: Remove the "For example....", it is pointing to ConnectionPolicy which is internal.

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 5]
```diff
@@ -197,16 +197,50 @@ public ConnectionMode ConnectionMode
...
+        /// </value>
         /// <remarks>
+        /// <para>
         /// The minimum interval is seconds. Any interval that is smaller will be ignored.
+        /// </para>
+        /// <para>
+        /// When a request fails due to a rate limiting error, the service sends back a response that
+        /// contains a value indicating the client should not retry before the <see cref="Microsoft.Azure.Documents.DocumentClientException.RetryAfter"/> time period has
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs:237]:
```suggestion
        /// contains a value indicating the client should not retry before the <see cref="Microsoft.Azure.Cosmos.CosmosException.RetryAfter"/> time period has
```

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 6]
```diff
@@ -197,16 +197,50 @@ public ConnectionMode ConnectionMode
...
+        /// elapsed. This property allows the application to set a maximum wait time for all retry attempts.
+        /// If the cumulative wait time exceeds the this value, the client will stop retrying and return the error to the application.
+        /// </para>
+        /// <para>
+        /// For more information, see <see href="https://docs.microsoft.com/azure/documentdb/documentdb-performance-tips#429">Handle rate limiting/request rate too large</see>.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs:242]:
```suggestion
        /// For more information, see <see href="https://docs.microsoft.com/azure/cosmos-db/performance-tips#429">Handle rate limiting/request rate too large</see>.
```

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 7]
```diff
@@ -197,16 +197,50 @@ public ConnectionMode ConnectionMode
...
+        /// the same request will be issued for a maximum of 10 times to the server before
+        /// an error is returned to the application. If the value of this property is set to 0,
+        /// there will be no automatic retry on rate limiting requests from the client and the exception
+        /// needs to handled at the application level.
+        /// For an example on how to set this value, please refer to <see cref="ConnectionPolicy.RetryOptions"/>.
+        /// </value>
+        /// <remarks>
+        /// <para>
+        /// When a client is sending requests faster than the allowed rate,
+        /// the service will return HttpStatusCode 429 (Too Many Request) to rate limit the client. The current
```
Comment (abhijitpai) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs:214]: nit: Requests plural - i.e. Too Many Requests

**Pull request merged**